File: languages/ocaml/test/x86.ml
open OUnit2

let suite = "x86 suite" >::: [
  "should assemble single ret instruction" >:: (fun ctx ->
    let buf = Iobuf.create 1 in

    X86.ret buf ;

    assert_equal ctx (Iobuf.to_string buf) "\xc3"
  );
];;

let () = run_test_tt_main suite ;;

File: languages/cpp/test/x86.cpp
#ifndef ASM_ALL_TESTS
  #define CATCH_CONFIG_MAIN
#endif

#include <sstream>
#include "catch.hpp"
#include "../src/x86.cpp"

using Catch::Matchers::Equals;

using namespace std::string_literals;

TEST_CASE("x86 tests", "[x86]") {
    std::ostringstream buf;

    SECTION("should assemble single ret instruction") {
        x86::ret(buf);

        REQUIRE( buf.tellp() == 1 );
        REQUIRE_THAT(buf.str(), Equals("\xc3"s));
    }
}
